app/models/spree/order_contents_decorator.rb

```diff
@@ -10,17 +31,34 @@ def add_to_line_item_with_parts(variant, quantity, options = {})
         )
       end
     end
+
     alias_method_chain :add_to_line_item, :parts
 
     private
 
+    def part_variants_match(line_item, variant, quantity, options)
+      if line_item.parts.any? && options["selected_variants"]
+        selected_variant_ids = options["selected_variants"].values.map(&:to_i)
+        matched = part_variant_ids(line_item) & selected_variant_ids
+
+        matched == selected_variant_ids
+      else
+        true
+      end
+    end
+
+    def part_variant_ids(line_item)
+      line_item.part_line_items.map(&:variant_id)
+    end
+
     def populate_part_line_items(line_item, parts, selected_variants)
       parts.each do |part|
-        line_item.part_line_items.create!(
+        part_line_item = line_item.part_line_items.find_or_initialize_by(
           line_item: line_item,
-          variant_id: variant_id_for(part, selected_variants),
-          quantity: part.count
+          variant_id: variant_id_for(part, selected_variants)
         )
+
+        part_line_item.update_attributes(quantity: part.count)
```

Comment:
Do we have something to handle this line if it return `false`? Should it be `update_attributes!`?